airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeDestinationHandler.kt

```diff
@@ -573,39 +575,49 @@ class SnowflakeDestinationHandler(
-                arrayOf(databaseName.uppercase(Locale.getDefault())) + namespaces + names
-            val results: List<JsonNode> = database.queryJsons(query, *bindValues)
-            for (result in results) {
-                val tableSchema = result["TABLE_SCHEMA"].asText()
-                val tableName = result["TABLE_NAME"].asText()
-                val columnName = result["COLUMN_NAME"].asText()
-                val dataType = result["DATA_TYPE"].asText()
-                val isNullable = result["IS_NULLABLE"].asText()
-                val tableDefinition =
-                    existingTables
-                        .computeIfAbsent(tableSchema) { _: String? -> LinkedHashMap() }
-                        .computeIfAbsent(tableName) { _: String? ->
-                            TableDefinition(LinkedHashMap())
-                        }
-                tableDefinition.columns[columnName] =
-                    ColumnDefinition(columnName, dataType, 0, fromIsNullableIsoString(isNullable))
+
+            try {
+                val existingTablesFromShowQuery =
+                    LinkedHashMap<String, LinkedHashMap<String, TableDefinition>>()
+                for (stream in streamIds) {
+                    val showColumnsQuery =
+                        """
+                        SHOW COLUMNS IN TABLE "$databaseName"."${stream.finalNamespace}"."${stream.finalName}";
+                        """.trimIndent()
+                    val showColumnsResult = database.queryJsons(
+                        showColumnsQuery,
+                    )
+                    for (result in showColumnsResult) {
+                        val tableSchema = result["schema_name"].asText()
+                        val tableName = result["table_name"].asText()
+                        val columnName = result["column_name"].asText()
+                        val dataType = changeDataTypeFromShowQuery(ObjectMapper().readTree(result["data_type"].asText()).path("type").asText())
+                        val isNullable = result["null?"].asText()
+                        val tableDefinition =
+                            existingTablesFromShowQuery
+                                .computeIfAbsent(tableSchema) { _: String? -> LinkedHashMap() }
+                                .computeIfAbsent(tableName) { _: String? ->
+                                    TableDefinition(LinkedHashMap())
+                                }
+                        tableDefinition.columns[columnName] =
+                            ColumnDefinition(
+                                columnName,
+                                dataType,
+                                0,
+                                isNullable.toBoolean()
+                            )
+                    }
+                }
+                return existingTablesFromShowQuery
+            } catch (e: SnowflakeSQLException) {
+                if(e.message != null && e.message!!.contains("does not exist")) {
```

Comment:
I'm not sure we should catch the exception outside of the loop that iterates over streams.